spyder/plugins/ipythonconsole/widgets/debugginghistory.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""
+Widget that manages a history file for the debugging.
+"""
+
+import pdb
+
+from qtconsole.rich_jupyter_widget import RichJupyterWidget
+from IPython.core.history import HistoryManager
+
+from spyder.config.base import get_conf_path
+
+
+class PdbHistory(HistoryManager):
+
+    def _get_hist_file_name(self, profile=None):
+        """
+        Get default pdb history file name.
+
+        The profile parameter is ignored, but must exist for compatibility with
+        the parent class.
+        """
+        return get_conf_path('pdb_history.sqlite')
+
+
+class DebuggingHistoryWidget(RichJupyterWidget):
```

Comment:
Why does this class need to inherit from `RichJupyterWidget`? I mean, I don't see it necessary because it's not overriding or changing anything from `RichJupyterWidget`.

Instead, I think it'd be clearer if this is a mixing and `DebuggingWidget` inherits from it and `RichJupyterWidget`. That would simplify the code because there wouldn't be a need to declare `is_debugging` here and redeclare it in `DebuggingWidget`.

Besides, I'd also suggest to move it to `widget/debugging.py` because all this code is related to additions to `RichJupyterWidget` for debugging. If you prefer it to leave it in a separate file, please move it to `utils/debugging.py`.

Reply:
it is overriding `_history`, `_history_edits`, and `_history_index`